kolibri/core/device/utils.py

```diff
@@ -75,3 +77,22 @@ def provision_device(**kwargs):
     device_settings, _ = DeviceSettings.objects.get_or_create(defaults=kwargs)
     device_settings.is_provisioned = True
     device_settings.save()
+
+
+def valid_app_key(app_key):
+    from .models import DeviceAppKey
+
+    return app_key == DeviceAppKey.get_app_key()
+
+
+def valid_app_key_on_request(request):
+    return APP_KEY_COOKIE_NAME in request.COOKIES and valid_app_key(
+        request.COOKIES.get(APP_KEY_COOKIE_NAME)
+    )
+
+
+def set_app_key_on_response(response):
+    from .models import DeviceAppKey
+
+    response.set_cookie(APP_KEY_COOKIE_NAME, DeviceAppKey.get_app_key())
+    return response
```

Comment:
Based on the function name, might be cleaner if it did the operation to the response but didn't return the response.